tools/rtpsrelay/RtpsRelay.cpp

```diff
@@ -873,12 +877,27 @@ int ACE_TMAIN(int argc, ACE_TCHAR* argv[])
   }
 
   RelayStatusReporter relay_status_reporter(config, guid_addr_set, relay_status_writer, reactor);
-  if (thread_mon.start() == -1) {
-    ACE_ERROR((LM_ERROR, ACE_TEXT("(%P:%t) ERROR: failed to activate Thread Load Monitor\n")));
-    return EXIT_FAILURE;
+
+  OpenDDS::DCPS::ThreadStatusManager& thread_status_manager = TheServiceParticipant->get_thread_status_manager();
+
+  if (thread_status_manager.update_thread_status()) {
+    if (relay_thread_monitor->start() == -1) {
+      ACE_ERROR((LM_ERROR, ACE_TEXT("(%P:%t) ERROR: failed to activate Thread Load Monitor\n")));
+      return EXIT_FAILURE;
+    }
+
+    OpenDDS::DCPS::ThreadStatusManager::Start s(thread_status_manager, "RtpsRelay Main");
+
+    for (;;) {
+      ACE_Time_Value t = thread_status_manager.thread_status_interval().value();
```

Comment:
```suggestion
      const ACE_Time_Value t = thread_status_manager.thread_status_interval().value();
```

Reply:
`run_reactor_event_loop` modifies the `ACE_Time_Value`.